lib/plugins/backup.rb

```diff
@@ -0,0 +1,125 @@
+require 'rubygems/package'
+require 'zlib'
+require_relative '../moonshot/creds_helper'
+
+module Moonshot::Plugins # rubocop:disable Documentation
+  # Monkeypatching TarWriter for our
+  # needs to improve clarity
+  refine Gem::Package::TarWriter do
+    def add(path:, name:, permission: nil)
+      rel_path = File.join(path << name)
+      permission ||= 0644
+
+      add_file(name, permission) do |io|
+        File.open(rel_path, 'r') { |f| io.write(f.read) }
+      end
+    end
+  end
+end
+
```

Comment:
@glennpratt: I went from monkey patching to refining, but the only way I could make this work was putting the refinement to the beginning, then say `using Moonshot::Plugins`. is this allright?